plugins/modules/ipauser.py

```diff
@@ -483,32 +484,26 @@
 def find_user(module, name, preserved=False):
     _args = {
         "all": True,
-        "uid": name,
     }
-    if preserved:
-        _args["preserved"] = preserved
-
-    _result = module.ipa_command("user_find", name, _args)
-
-    if len(_result["result"]) > 1:
-        module.fail_json(
-            msg="There is more than one user '%s'" % (name))
-    elif len(_result["result"]) == 1:
-        # Transform each principal to a string
-        _result = _result["result"][0]
-        if "krbprincipalname" in _result \
-           and _result["krbprincipalname"] is not None:
-            _list = []
-            for x in _result["krbprincipalname"]:
-                _list.append(str(x))
-            _result["krbprincipalname"] = _list
-        certs = _result.get("usercertificate")
-        if certs is not None:
-            _result["usercertificate"] = [encode_certificate(x)
-                                          for x in certs]
-        return _result
-
-    return None
+
+    try:
+        _result = module.ipa_command("user_show", name, _args)
+    except ipalib_errors.NotFound:
+        return None
+
+    # Transform each principal to a string
+    _result = _result["result"]
+    if "krbprincipalname" in _result \
+       and _result["krbprincipalname"] is not None:
+        _list = []
+        for x in _result["krbprincipalname"]:
+            _list.append(str(x))
+        _result["krbprincipalname"] = _list
```

Comment:
What about `_result["krbprincipalname"] = [to_text(x) for x in _result["krbprincipalname"]]` instead?

Reply:
Fixed.
I also modified the handling of 'usercertificate' so both attributes use similar logic.